test/widgets/inbox_test.dart

```diff
@@ -385,6 +426,30 @@ void main() {
           checkAppearsUncollapsed(tester, 1, findSectionContent);
         });
 
+        testWidgets('collapse stream section after scroll', (tester) async {
```

Comment:
Let's also have a test that checks that when you collapse a section while the header is in the middle of the screen, we *don't* scroll to put it at the top. It was nontrivial to get that right:
https://github.com/zulip/zulip-flutter/pull/560#discussion_r1521885523
which makes it an especially good target for a test.